src/main/webapp/bookScript.js

```diff
@@ -17,7 +17,9 @@ document.addEventListener('DOMContentLoaded', () => {
           fetch(URL).then(response => {
             if (response.status === 200) {
               response.json().then(result => {
-                addBooksGrid(result.items);
+                if (extractBookList(result) !== null) {
+                  addBooksGrid(extractBookList(result));
+                }
```

Comment:
No need to call this twice :)

```suggestion
                const bookList = extractBookList(result);
                if (bookList !== null) {
                  addBooksGrid(bookList);
                }
```

Reply:
Fixed it